### PR TITLE
feat(dynamodb): add DynamoDB adapter

### PR DESCRIPTION
**Pull request by PolMrt**
I've create an adapter for DynamoDB. This follow the one table structure. You can find more info on the readme about the table structutre.

### CODE REVIEW

File: packages/dynamodb/package.json [hunk 1]
```diff
@@ -0,0 +1,27 @@
+{
+  "name": "@next-auth/example-adapter",
```
Comment (ibrahimcesar) [inline packages/dynamodb/package.json:2]: Great job! I stopped work on the implementation, I'll test soon yours. Just update this little left over in order to the package have a proper name on npm.

Comment (PolMrt) [reply packages/dynamodb/package.json]: Thanks for the heads up ! I've updated the name :) Let me know if you catch any bug while testing

File: packages/dynamodb/index.js [hunk 2]
```diff
@@ -0,0 +1,620 @@
...
+      const userId = randomBytes(16).toString("hex");
+      const now = new Date();
+      let item = {
+        pk: `USER#${userId}`,
+        sk: `USER#${userId}`,
+        id: userId,
+        type: "USER",
+        name: profile.name,
+        email: profile.email,
+        iamge: profile.image,
```
Comment (ibrahimcesar) [inline packages/dynamodb/index.js:50]: I believe is just a minor typo, `iamge` should be `image`

Comment (PolMrt) [reply packages/dynamodb/index.js]: Indeed, thanks :)

**Review by balazsorban44 — APPROVED** [review]
Thank you! After a long pause, we are getting back to the adapters repo! I am going to merge this for now!

**Pull request merged**